dds/DCPS/DiscoveryBase.h

```diff
@@ -779,7 +779,7 @@ namespace OpenDDS {
             DCPS::WriterIdSeq writer_seq(1);
             writer_seq.length(1);
             writer_seq[0] = removing;
-            lsi->second.remote_opendds_associations_.erase(removing);
+            lsi->second.remote_expectant_opendds_associations_.erase(removing);
```

Comment:
If this data isn't actually changing, the call to `write_subscription_data` below should be skipped.